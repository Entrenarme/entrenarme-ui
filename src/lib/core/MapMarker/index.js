// @flow
import * as React from 'react';
import styled from 'styled-components';

import { RESOURCES_URL } from '../../helpers/config';

const MapMarkerContainer = styled.div`
  width: 40px;
  height: 50px;
  cursor: pointer;
  background-image: ${props =>
    `url(${`${RESOURCES_URL}static/images/list/marker-${props.options.type}-${
      props.options.visited ? 'visited' : 'default'
    }.svg`})`};

  &:hover {
    background-image: ${props =>
      `url(${`${RESOURCES_URL}static/images/list/marker-${props.options.type}${
        props.options.visited ? '-visited' : ''
      }-hover.svg`})`};
  }

  @media only screen and (max-width: 767px) {
    width: 32px;
    height: 40px;
  }
`;

type Props = {
  type: 'trainer' | 'center',
  visited?: boolean,
};

const MapMarker = ({ type, visited }: Props) => (
  <MapMarkerContainer options={{ type, visited }} />
);

MapMarker.defaultProps = {
  type: 'trainer',
  visited: false,
};

export default MapMarker;
