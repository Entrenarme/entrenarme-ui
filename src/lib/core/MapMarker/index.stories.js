// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import MapMarker from './index';

storiesOf('MapMarker', module).add(
  'default',
  withInfo('Map marker')(() => <MapMarker />),
);

storiesOf('MapMarker', module).add(
  'center type',
  withInfo('Map marker')(() => <MapMarker type="center" />),
);

storiesOf('MapMarker', module).add(
  'trainer type',
  withInfo('Map marker')(() => <MapMarker type="trainer" />),
);

storiesOf('MapMarker', module).add(
  'center type visited',
  withInfo('Map marker')(() => <MapMarker type="center" visited />),
);
