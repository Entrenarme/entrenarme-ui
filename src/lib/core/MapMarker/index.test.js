import * as React from 'react';
import { render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';

import MapMarker from './index';

test('renders "MapMarker" with no props', () => {
  const { container } = render(<MapMarker />);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders "Map Marker" with type center', () => {
  const { container } = render(<MapMarker type="center" />);

  expect(container.firstChild).toMatchSnapshot();
});

test('renders "Map Marker" with type center and visited', () => {
  const { container } = render(<MapMarker type="center" visited />);

  expect(container.firstChild).toMatchSnapshot();
});
