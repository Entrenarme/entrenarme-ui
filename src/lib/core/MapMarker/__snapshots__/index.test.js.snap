// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "Map Marker" with type center 1`] = `
<div
  class="sc-bdVaJa gdRYsL"
/>
`;

exports[`renders "Map Marker" with type center and visited 1`] = `
<div
  class="sc-bdVaJa bCUWMx"
/>
`;

exports[`renders "MapMarker" with no props 1`] = `
<div
  class="sc-bdVaJa efdQpK"
/>
`;
